### PR TITLE
[#9255] Migrate part of instructor feedback sessions page to Angular

### PR DESCRIPTION
**Pull request by xpdavid**
Part of #9255

teammates online peer feedback evaluation system for student team projects 5

Some components added are also shared by instructor home page also. I added them to the instructor home page.

teammates online peer feedback evaluation system for student team projects 6

@jacoblipech Less work for you.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
@xpdavid need to resolve the conflicts.

File: src/main/java/teammates/ui/newcontroller/FeedbackSessionInfo.java [hunk 1]
```diff
@@ -380,10 +432,32 @@ public void validate() {
     }
 
     /**
-     * The basic request body format for saving of feedback session.
+     * The request body format for saving of feedback session.
      */
     public static class FeedbackSessionSaveRequest extends FeedbackSessionBasicRequest {
```
Comment (wkurniawan07) [inline src/main/java/teammates/ui/newcontroller/FeedbackSessionInfo.java]: If by now this has not been implemented, what is the usage of this?

File: src/web/app/components/copy-session-modal/copy-session-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    <i class="fas fa-times"></i>
+  </button>
+</div>
+<div class="modal-body">
+  <div class="row">
+    <div class="col-12">
+      <div class="form-group">
+        <label><b>Name for copied session</b></label>
+        <input type="text" class="form-control" [(ngModel)]="newFeedbackSessionName" maxlength="38">
+        <span>{{ 38 - newFeedbackSessionName.length }} characters left</span>
```
Comment (wkurniawan07) [inline src/web/app/components/copy-session-modal/copy-session-modal.component.html]: Magic number `38`?

File: src/web/app/components/sessions-table/submission-status-tooltip.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+   */
+  transform(status: FeedbackSessionSubmissionStatus): string {
+    let msg: string = 'The feedback session has been created';
+
+    switch (status) {
+      case FeedbackSessionSubmissionStatus.VISIBLE_NOT_OPEN:
+      case FeedbackSessionSubmissionStatus.OPEN:
+      case FeedbackSessionSubmissionStatus.GRACE_PERIOD:
+      case FeedbackSessionSubmissionStatus.CLOSED:
+        msg += 'The feedback session has been created, is visible';
```
Comment (wkurniawan07) [inline src/web/app/components/sessions-table/submission-status-tooltip.pipe.ts]: Will this not cause duplicate `The feedback session has been created`?

File: src/web/app/pages-instructor/instructor-session-base-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+  sendRemindersToStudents(model: SessionsTableRowModel): void {
+    const paramMap: { [key: string]: string } = {
+      courseid: model.feedbackSession.courseId,
+      fsname: model.feedbackSession.feedbackSessionName,
+    };
+
+    this.httpRequestService.post('/session/remind/submission', paramMap).subscribe(() => {
+      this.statusMessageService.showSuccessMessage(
+          'Reminder e-mails have been sent out to those students and instructors. '
+          + 'Please allow up to 1 hour for all the notification emails to be sent out..');
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-session-base-page.component.ts]: Duplicate `.` at the end.

File: src/web/services/template-sessions.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+ */
+export interface TemplateSession {
+  name: string;
+  questions: FeedbackQuestion[];
+}
+
+/**
+ * Template sessions.
+ */
+export const templateSessions: TemplateSession[] = [
```
Comment (wkurniawan07) [inline src/web/services/template-sessions.ts]: This should be a JSON file instead? This is a form of data, not application code.

**Comment by xpdavid** [comment]
> Also, it looks like the sort table function is buggy? When I tried, only one column (i.e. where I pressed the icon) gets sorted.

Only one sorting criteria is applied each time  (i.e. cannot sort by Course ID first and then Session Name). That is also the functionality in V6.

File: src/main/java/teammates/ui/newcontroller/FeedbackSessionInfo.java (hunk 1 above)
Comment (xpdavid) [reply src/main/java/teammates/ui/newcontroller/FeedbackSessionInfo.java]: It is used in `SaveFeedbackSessionAction`. I just fix the typo I introduced in the previous PR :P

File: src/web/services/template-sessions.ts (hunk 5 above)
Comment (xpdavid) [reply src/web/services/template-sessions.ts]: I need `interface TemplateSession {}` defined somewhere. So I thought put them together would be more organised.

Comment (wkurniawan07) [reply src/web/services/template-sessions.ts]: More organized is one thing. The issue here is you're mixing up application code and application data (independent of logic).

Comment (xpdavid) [reply src/web/services/template-sessions.ts]: Ok. One problem of serialised it to JSON file is that we will loss reference to enum type (e.g. `FeedbackVisibilityType.RECIPIENT` will become `"RECIPIENT"`) Not sure whether it is a bit problem or not for project evolution.

**Pull request merged**